assets/js/util.js

```diff
@@ -242,8 +243,29 @@ function synchronizeTextareaHeights() {
     $('#translatedText').css('height', originalTextScrollHeight + 'px');
 }
 
-/*:: export {synchronizeTextareaHeights, modeEnabled, ajaxSend, ajaxComplete, filterLangList, onlyUnique,
-    SPACE_KEY_CODE, ENTER_KEY_CODE, HTTP_OK_CODE, HTTP_BAD_REQUEST_CODE, XHR_LOADING, XHR_DONE} */
+function ajaxCallForAllModes(methodType, request, endpoint, callbackSuccess, callbackError, requestName) {
+    requestName = $.ajax({
+        url: config.APY_URL + endpoint,
+        beforeSend: ajaxSend,
+        complete: function () {
+            ajaxComplete();
+            requestName = undefined;
+        },
+        data: request,
+        type: methodType,
+        contentType: 'application/x-www-form-urlencoded; charset=UTF-8',
+        dataType: 'json',
+        success: function (data) {
+            callbackSuccess(data);
+        },
```

Comment:
> Instead of making each of these separate options passed through the arguments, you could make it accept an options Object and merge that Object with the one here. That would be the most extensible solution. Then the call sites for this function would provide their input as an object (endpoint can remain a separate argument).

@sushain97 , Um, do you mean something like this?

```
    var translateObject = {
        'methodType': methodType,
        'request': request,
        'success': success,
        'error': error,
        'requestName': 'textTranslateRequest'
    };
    apyCall(translateObject, endpoint);
```

And then the apyCall would look something like this?

```
function apyCall(requestObject, endpoint) {
    // code for checking the method type would be here instead of in the object
    textTranslateRequest = $.ajax({
        url: config.APY_URL + endpoint,
        beforeSend: ajaxSend,
        complete: function () {
            ajaxComplete();
            textTranslateRequest = undefined;
        },
        data: requestObject.request,
        type: requestObject.methodType,
        contentType: 'application/x-www-form-urlencoded; charset=UTF-8',
        dataType: 'json',
        success: requestObject.success,
        error: requestObject.error
    });
}

```

Here, each of the `success` and `error` callbacks have been defined before passing them into the request object. Finally, we pass this object as an argument to the `apyCall()`.